packages/console/src/browser/console-session-manager.ts

```diff
@@ -0,0 +1,107 @@
+/********************************************************************************
+ * Copyright (C) 2021 TypeFox and others.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * This Source Code may also be made available under the following Secondary
+ * Licenses when the conditions for such availability set forth in the Eclipse
+ * Public License v. 2.0 are satisfied: GNU General Public License, version 2
+ * with the GNU Classpath Exception which is available at
+ * https://www.gnu.org/software/classpath/license.html.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+ ********************************************************************************/
+
+import { injectable } from '@theia/core/shared/inversify';
+import { Emitter, Disposable, DisposableCollection } from '@theia/core';
+import { ConsoleSession } from './console-session';
+import { Severity } from '@theia/core/lib/common/severity';
+
+@injectable()
+export class ConsoleSessionManager implements Disposable {
+
+    protected readonly sessions = new Map<string, ConsoleSession>();
+    protected _selectedSession: ConsoleSession | undefined;
+    protected _severity: Severity | undefined;
+
+    protected readonly sessionAddedEmitter = new Emitter<ConsoleSession>();
+    protected readonly sessionDeletedEmitter = new Emitter<ConsoleSession>();
+    protected readonly sessionWasShownEmitter = new Emitter<ConsoleSession>();
+    protected readonly sessionWasHiddenEmitter = new Emitter<ConsoleSession>();
+    protected readonly selectedSessionChangedEmitter = new Emitter<ConsoleSession | undefined>();
+    protected readonly severityChangedEmitter = new Emitter<void>();
+
+    readonly onSessionAdded = this.sessionAddedEmitter.event;
```

Comment:
Given the [coding guidelines](https://github.com/eclipse-theia/theia/wiki/Coding-Guidelines#event_names) we should follow the following format: `on[Will|Did]VerbNoun`.

```suggestion
    readonly onDidAddSession = this.sessionAddedEmitter.event;
```

The same is true for other event names.

Reply:
Thanks! I actually didn't know about that. Done.